src/main/java/ch/uzh/ifi/hase/soprafs23/service/LobbyService.java

```diff
@@ -265,4 +265,16 @@ public int getRemainingPlayersNeeded(int accessCode) {
         }
         return remainingPlayers;
     }
+
+    public void deleteUsersNotInTeam(int accessCode){
+        Lobby lobby = lobbyRepository.findByAccessCode(accessCode);
+        List<User> allUsers = lobby.getLobbyUsers();
+        for(int i = 0; i<allUsers.size(); i++){
+            if(!lobby.isUserInTeam1(allUsers.get(i)) && !lobby.isUserInTeam2(allUsers.get(i))){
+                leaveLobby(accessCode, Math.toIntExact(allUsers.get(i).getId()));
```

Comment:
I don't understand this, but I assume it works.

Reply:
it goes over all users in the lobby and if they aren't in a team they will be removed, it works